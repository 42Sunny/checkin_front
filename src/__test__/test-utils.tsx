// test-utils.jsx
import { configureStore } from "@reduxjs/toolkit";
import { Queries, render, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement } from "react";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
// Import your own reducer
import renderer from "react-test-renderer";
import appStore from "../redux/configureStore";
import rootReducer, { RootState } from "../redux/modules";
import { initialState as clusterInitState } from "../redux/modules/cluster";
import { initialState as userInitState } from "../redux/modules/user";

const makeCustomStore = ({
  cluster,
  user,
}: {
  cluster?: typeof clusterInitState;
  user?: typeof userInitState;
} = {}) => {
  return {
    clusterReducer: { ...clusterInitState, ...cluster },
    userReducer: { ...userInitState, ...user },
  };
};

const customRender = (
  ui: ReactElement,
  preloadedState: RootState = makeCustomStore(),
  { store = configureStore({ reducer: rootReducer, preloadedState }), ...renderOptions } = {},
  history = createMemoryHistory(),
): RenderResult<Queries, HTMLElement> => {
  const Wrapper: any = ({ children }: { children: ReactElement }) => {
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

const createWithWrapper = (children: ReactElement) =>
  renderer.create(
    <Provider store={appStore}>
      <Router history={createMemoryHistory()}>{children}</Router>
    </Provider>,
  );

// re-export everything
export * from "@testing-library/react";
// override render method
export { customRender as render, createWithWrapper, makeCustomStore };
