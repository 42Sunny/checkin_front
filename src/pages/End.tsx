import React, { useEffect } from "react";
import { useHistory } from "react-router-dom";
import classes from "../styles/pages/EndPage.module.css";

// TODO: 뒤로가기할때 이 페이지가 나옴 이거 없애고 모달로 만드ㄹ
const End = () => {
  const history = useHistory();

  useEffect(() => {
    const timeOutId = setTimeout(() => history.push("/checkin"), 1000);
    return () => {
      clearTimeout(timeOutId);
    };
  }, [history]);

  return (
    <div className={classes["text-wrapper"]}>
      <h1 className={classes["ending-text"]}>Complete!</h1>
    </div>
  );
};

export default End;
