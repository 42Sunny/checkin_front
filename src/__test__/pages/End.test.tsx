import { findByRole, waitFor } from "@testing-library/react";
import { createMemoryHistory } from "history";
import End from "../../pages/End";
import { act, render, screen, createWithWrapper } from "../test-utils";

describe("src/pages/End.tsx", () => {
  it("matches snapshot", () => {
    const component = createWithWrapper(<End />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders checkinPage after 1 second", async () => {
    // TODO:
    // const history = createMemoryHistory();
    // render(<End />, undefined, undefined, history);
    // await waitFor(() => expect(history.length).toBe(2));
    // console.log(history);
    // screen.debug();
  });
});
