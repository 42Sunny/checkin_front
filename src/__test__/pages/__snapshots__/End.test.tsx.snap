// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/End.tsx matches snapshot 1`] = `
<div
  className="text-wrapper"
>
  <h1
    className="ending-text"
  >
    Complete!
  </h1>
</div>
`;
